### PR TITLE
Improve event handling for nested tabs

### PR DESCRIPTION
**Pull request by dbatiste**
The original code would have worked - this change just avoid possible JS error when `d2l-tab-panel-selected` or `d2l-tab-panel-text-changed` is dispatched from a nested set of tabs.

### CODE REVIEW

File: components/tabs/demo/tabs.html [hunk 1]
```diff
@@ -118,6 +118,27 @@ <h3>Tabs (max-to-show)</h3>
...
+								<d2l-tab-panel text="Geochemistry">Tab content for Geochemistry</d2l-tab-panel>
+								<d2l-tab-panel text="Groundwater Modeling">Tab content for Groundwater Modeling</d2l-tab-panel>
+								<d2l-tab-panel text="Stratigraphy">Tab content for Stratigraphy</d2l-tab-panel>
+							</d2l-tabs>
+						</d2l-tab-panel>
+						<d2l-tab-panel text="Physics">Tab content for Physics</d2l-tab-panel>
+					</d2l-tabs>
+				</template>
+			</d2l-demo-snippet>
+
```
Comment (dlockhart) [inline components/tabs/demo/tabs.html:141]: Personally I don't like it when really weird edge cases that we don't expect to see often "in the wild" take up space in our demo pages -- which are meant to be a showcase of the component.

How hard would it be to move this into a test that exercises the error and now the solution?

Comment (dbatiste) [reply components/tabs/demo/tabs.html:141]: I'm am tempted to just ignore it in that case.

Comment (svanherk) [reply components/tabs/demo/tabs.html]: Yeah I could write one in my-courses, but I'm also working to get rid of the nesting there since it's not necessary.  And then I don't know of anywhere else that uses it... maybe if someone has put a filter dropdown inside a tab panel somewhere?
